WEB/Src/PerformanceCollector/Perf.Tests/QuickPulse/QuickPulseServiceClientTests.cs

```diff
@@ -1208,17 +1238,47 @@ public void QuickPulseServiceClientSubmitsInstanceNameToServiceWithPing()
             Assert.AreEqual(1, this.pings.Count);
             Assert.AreEqual(instanceName, this.pings[0].Item1.InstanceName);
             Assert.AreEqual(instanceName, this.pings[0].Item3.Instance);
+            Assert.AreEqual(roleName, this.pings[0].Item3.RoleName);
         }
 
         [TestMethod]
-        public void QuickPulseServiceClientSubmitsInstanceNameToServiceWithSubmitSamples()
+        public void QuickPulseServiceClientSubmitsRoleNameWithNullValueValidation()
+        {
+            // ARRANGE
+            var serviceClient = new QuickPulseServiceClient(
+                this.TestContext.Properties[ServiceEndpointPropertyName] as Uri,
+                string.Empty,
+                null,
+                string.Empty,
+                string.Empty,
+                string.Empty,
+                new Clock(),
+                false,
+                0);
+
+            // ACT
+            CollectionConfigurationInfo configurationInfo;
+            serviceClient.Ping(Guid.NewGuid().ToString(), DateTimeOffset.UtcNow, string.Empty, string.Empty, out configurationInfo);
+
+            // SYNC
```

Comment:
[minor] ARRANGE - ACT - ASSERT is the constant format here, waiting should probably be under ASSERT.